lib/vrf.c

```diff
@@ -582,10 +582,9 @@ int vrf_socket(int domain, int type, int protocol, vrf_id_t vrf_id,
 			     __func__, vrf_id, safe_strerror(errno));
 
 	if (ret > 0 && interfacename && vrf_default_accepts_vrf(type)) {
-		zlog_err("VRF socket not used since net.ipv4.%s_l3mdev_accept != 0",
-			  (type == SOCK_STREAM ? "tcp" : "udp"));
-		errno = EEXIST; /* not sure if this is the best error... */
-		return -2;
+		zlog_warn("Using default rather than VRF socket for %s since net.ipv4.%s_l3mdev_accept != 0",
+			  interfacename, (type == SOCK_STREAM ? "tcp" : "udp"));
+		interfacename = NULL;
```

Comment:
I confirm this solves the problem Donald reported on #3720.

However, since we can't stop using `SO_REUSEADDR`, `bind()` will never fail (because we're also using `SO_REUSEPORT`), which leave us with multiple open sockets listening to the same port. While that works, it's not the ideal solution to have.

Removing only the call to `sockopt_reuseport()` should fix that issue (multiple listening sockets). Only problem is that the bgpd log files will be flooded with lots of `bind: Address already in use` warnings when multiple VRFs are configured, and this can be a source of concern and confusion to the FRR users. A better solution would probably involve some level of refactoring in the VRF code to implement socket refcouting when `tcp_l3mdev_accept` is enabled.